### DISCUSSION TITLE
is hovering support for hyphen supported variables for cobol?

### DISCUSSION THREAD

**Discussion by teju242** [IBM/zopeneditor-about#216] [Q&A]
I am trying to get values on hovering variable. I have variables with hyphen in name.
E.g.- EMP-SALARY.

**Comment by phaumer** [comment]
Hello @teju242.
Do you mean hovers such as these? 
<img width="356" alt="2022-02-11_10-15-07" src="https://user-images.githubusercontent.com/4391934/153647399-9093cf69-da36-435a-9176-6cbd525dc09e.png">

If you do not see them then most often it is a problem with finding Java for running the language server. Does the Z Open Editor Welcome page tell you that it has found a valid version of Java to use?

**Comment by FALLAI-Denis** [comment]
Hi @teju242 

What do you mean by "value"?
If it's about displaying the value held in a variable then I think you're referring to a debugger, which Z Open Editor is not.

It is also possible that information when hovering over an area is not displayed if the language server is paused.

**Comment by phaumer** [comment]
That is correct. If you are using our IBM Z Open Debug VS Code extension then that one does not yet show values in hovers. You can see them in the Variables view on the left of the Debug activity bar.

**Reply by teju242** [reply]
Yes.  IBM Z Open Debug has known issue and it is mentioned here https://www.ibm.com/support/pages/known-issues-and-limitations-ibm-z-open-debug-wazi-developer-visual-studio-code-and-wazi-developer-workspaces-ides.

**Comment by teju242** [comment]
Thank you all.
z-IBM editor has known issue. It is already mentioned here https://www.ibm.com/support/pages/known-issues-and-limitations-ibm-z-open-debug-wazi-developer-visual-studio-code-and-wazi-developer-workspaces-ides.

**Reply by FALLAI-Denis** [reply]
Issues for Z Open Debug, not for Z Open Editor...

**Reply by teju242** [reply]
Yes. My bad.
